source/base/ByteBuffer.ooc

```diff
@@ -85,12 +80,10 @@ _SlicedByteBuffer: class extends ByteBuffer {
 	}
 }
 _RecoverableByteBuffer: class extends ByteBuffer {
-	_recover: Func (ByteBuffer)
+	_recover: Func (ByteBuffer) -> Bool
 	init: func (pointer: UInt8*, size: Int, =_recover) { super(pointer, size) }
 	free: override func {
-		if (!this _forceFree)
-			this _recover(this)
-		else {
+		if (this _recover(this)) {
```

Comment:
You are right, the logic should be inverted. If `recover` succeeded it should return `true` and that should stop the freeing. @emilwestergren can you fix this before it goes of to merging?